examples/guides/mp-restful-webservice/src/main/docs/index.adoc

```diff
@@ -0,0 +1,487 @@
+
+    Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+:adoc-dir: {guides-dir}/mp-restful-webservice/src/main/docs
+:java-base: {adoc-dir}/../java/io/helidon/guides/mp/restfulwebservice
+:greet-app: {java-base}/GreetApplication.java
+:main-class: {java-base}/Main.java
+:greet-resource-class: {java-base}/GreetResource.java
+:greeting-message-class: {java-base}/GreetingMessage.java
+:greet-application-class: {java-base}/GreetApplication.java
+:health-resource-class: {java-base}/HealthResource.java
+:pom: {adoc-dir}/../../../pom.xml
+:meta-inf-dir: {adoc-dir}/../resources/META-INF
+:beans-xml: {meta-inf-dir}/beans.xml
+:config-file: {meta-inf-dir}/microprofile-config.properties
+:logging-properties-file: {adoc-dir}/../resources/logging.properties
+:se-guide-adoc: {guides-dir}/se-restful-webservice/src/main/docs/index.adoc
+
+
+= The MP RESTful Web Service Guide
+:description: Helidon MP guide restful web service
+:keywords: helidon, guide, example, mp
+:toc:
+
+Create and build a JAX-RS RESTful web service as your first Helidon MP application.
+
+== What You Will Learn
+You'll learn how to use Helidon MP quickly to create a JAX-RS RESTful web service that accepts these HTTP requests:
+
+|===
+|Method and URL | Result
+
+|`GET localhost:8080/greet` |Returns a generic but friendly greeting
+|`GET localhost:8080/greet/Joe` |Returns a personalized greeting for the specified person
+|`PUT localhost:8080/greet/greeting/Hola` |Changes the greeting used in subsequent responses
+|===
+
+We create the app in these main steps:
+
+. Write a basic Helidon MP app to respond to the HTTP requests.
+. Add code to perform simple health checks.
+. Add code to record simple metrics.
+
+(See the last section to find out how to get the entire finished code for this example.)
+
+== What You Need
+
+|===
+|About 15 minutes
+|An IDE or text editor
+|JDK 1.8 or later
+|Maven 3.5 or later
+|===
+
+//Optional:
+//|===
+//|Docker 18.02 (use the Edge channel to run Kubernetes on your desktop)
+//|`kubectl` 1.7.4
+//|===
+
+== Create Your Application
+
+=== Create a new Maven Project
+You can create your maven project in these ways:
+
+* use your IDE to create a new Java maven project, or
+* run the standard maven archetype to create a new Java project using this command:
++
+[source,bash]
+----
```

Comment:
CONTENT: To enable users running Maven on Windows to copy and paste the archetype:generate commands at the CMD prompt, consider showing the entire command in one line in the doc, i.e., without the \ line separators.

Reply:
CONTENT: Add an instruction to enter the groupID, artifactId, version, and package when prompted.

Reply:
About the backslashes... An explicit recommendation from an earlier review was to _add in_ the backslashes and break the example onto separate lines because the single-line version did not wrap -- it would scroll horizontally -- and therefore be awkward for the user to view all of it. 

That said, you raise a good point. One option (despite the mild maintenance pain it would induce) would be to have two code examples for long commands: one Unix-style and one Windows (using the ^ as the continuation character). 

I think this `mvn` command is the only one in this guide long enough to cause wrapping/scrolling issues.

If we think this is a good idea then for consistency we probably need to be prepared to adopt it in all content, not just the guides, on our site where we display long commands.

Thoughts?

Reply:
> have two code examples for long commands: one Unix-style and one Windows (using the ^ as the continuation character).

I agree.

Reply:
To be honest, our guides are going to have a Unix bias. For example the Dockerfile guide I wrote uses `du`, `ls`, etc. And I'm hesitant to harm readability/simplicity by always including Windows syntax. I'm OK with 80% solutions.

But I do see how it would be good for these long mvn archetype commands. Tim, do you think we could arrange the two snippets side-by-side? That might look OK and not overly clutter the document.

Reply:
I'll look into a side-by-side display.

Reply:
Side by side is probably not a good idea.
Having them in a list would make more sense.

Reply:
Done as a list.